crates/common/logged_command/src/logged_command.rs

```diff
@@ -148,23 +148,34 @@ impl std::fmt::Display for LoggedCommand {
 }
 
 impl LoggedCommand {
-    pub fn new(program: impl AsRef<OsStr>) -> LoggedCommand {
+    pub fn new(program: impl AsRef<OsStr>) -> Result<LoggedCommand, shell_words::ParseError> {
         let command_line = match program.as_ref().to_str() {
             None => format!("{:?}", program.as_ref()),
             Some(cmd) => cmd.to_string(),
         };
 
-        let mut command = Command::new(program);
+        let mut args = shell_words::split(&command_line)?;
+
+        let mut command = match args.len() {
+            0 => Command::new(program),
```

Comment:
This should return an `Err(shell_words::ParseError)`, as this occurs with an empty command line.

Reply:
I was thinking about that, though in previous implementation user was allowed to pass empty command line so I wanted to preserve that behavior.